AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -207,30 +214,54 @@ - (void)recordMessageOpenedEventForNotification:(NSDictionary *) userInfo
 }
 
 #pragma mark - Helpers
-- (void)addCampaignMetadataForEvent:(AWSPinpointEvent *) event
+- (void)addPinpointMetadataForEvent:(AWSPinpointEvent *) event
                    withNotification:(NSDictionary *) userInfo {
-    if (![AWSPinpointNotificationManager validCampaignPushForNotification:userInfo]) {
+    if (![AWSPinpointNotificationManager validPinpointPushForNotification:userInfo]) {
         return;
     }
-    NSDictionary *campaign = userInfo[AWSDataKey][AWSPinpointKey][AWSPinpointCampaignKey];
-    AWSDDLogVerbose(@"Adding campaign attributes to event[%@]: %@",event.eventType, campaign);
     
-    for (NSString *key in [campaign allKeys]) {
-        [event addAttribute:campaign[key] forKey:key];
+    NSDictionary *pinpointData = userInfo[AWSDataKey][AWSPinpointKey];
+    if (pinpointData[AWSPinpointCampaignKey]) {
+        NSDictionary *campaign = pinpointData[AWSPinpointCampaignKey];
+        AWSDDLogVerbose(@"Adding campaign attributes to event[%@]: %@", event.eventType, campaign);
+
+        for (NSString *key in [campaign allKeys]) {
+            [event addAttribute:campaign[key] forKey:key];
+        }
+    } else if (pinpointData[AWSPinpointJourneyKey]) {
+        NSDictionary *journey = pinpointData[AWSPinpointJourneyKey];
+        AWSDDLogVerbose(@"Adding journey attributes to event[%@]: %@", event.eventType, journey);
+
+        for (NSString *key in [journey allKeys]) {
+            [event addAttribute:journey[key] forKey:key];
+        }
+    } else {
+        AWSDDLogVerbose(@"No valid Pinpoint Push payload found");
     }
 }
 
-- (void)addGlobalCampaignMetadataForNotification:(NSDictionary *) userInfo {
-    if (![AWSPinpointNotificationManager validCampaignPushForNotification:userInfo]) {
+- (void)addGlobalPinpointMetadataForNotification:(NSDictionary *) userInfo {
+    if (![AWSPinpointNotificationManager validPinpointPushForNotification:userInfo]) {
         return;
     }
-    
-    NSDictionary *campaign = userInfo[AWSDataKey][AWSPinpointKey][AWSPinpointCampaignKey];
-    AWSDDLogVerbose(@"Adding campaign global attributes: %@", campaign);
-    [self.context.analyticsClient setCampaignAttributes:campaign];
-    
-    for (NSString *key in [campaign allKeys]) {
-        [self.context.analyticsClient addGlobalAttribute:campaign[key] forKey:key];
+
+    NSDictionary *pinpointData = userInfo[AWSDataKey][AWSPinpointKey];
+    if (pinpointData[AWSPinpointCampaignKey]) {
```

Comment:
1. It's not possible, one notification cannot contain attributes for both.
2. I was hesitant to do this because of exactly what you said. But it does make those two method a lot cleaner. I will make this change.